python/huggingfaceserver/huggingfaceserver/vllm_completions.py

```diff
@@ -0,0 +1,452 @@
+            prompt_logprobs = final_res.prompt_logprobs
+            prompt_text = final_res.prompt
+
+            for output in final_res.outputs:
+                if request.echo and request.max_tokens == 0:
+                    token_ids = prompt_token_ids
+                    top_logprobs = prompt_logprobs
+                    output_text = prompt_text
+                elif request.echo and request.max_tokens > 0:
+                    token_ids = prompt_token_ids + output.token_ids
+                    top_logprobs = prompt_logprobs + output.logprobs
+                    output_text = prompt_text + output.text
+                else:
+                    token_ids = output.token_ids
+                    top_logprobs = output.logprobs
+                    output_text = output.text
+
+                if request.logprobs is not None:
+                    logprobs = self._create_logprobs(
+                        token_ids=token_ids,
+                        top_logprobs=top_logprobs,
+                        num_output_top_logprobs=request.logprobs,
+                    )
+                else:
+                    logprobs = None
+
+                choice_data = CompletionChoice(
+                    index=len(choices),
+                    text=output_text,
+                    logprobs=logprobs,
+                    finish_reason=output.finish_reason,
+                    stop_reason=output.stop_reason,
+                )
+                choices.append(choice_data)
+
+            num_prompt_tokens += len(prompt_token_ids)
+            num_generated_tokens += sum(
+                len(output.token_ids) for output in final_res.outputs
+            )
+
+        usage = CompletionUsage(
+            prompt_tokens=num_prompt_tokens,
+            completion_tokens=num_generated_tokens,
+            total_tokens=num_prompt_tokens + num_generated_tokens,
+        )
+
+        return Completion(
+            id=request_id,
+            created=created_time,
+            model=model_name,
+            object="text_completion",
+            choices=choices,
+            usage=usage,
+        )
+
+    def apply_chat_template(self, messages: Iterable[ChatCompletionMessageParam]):
+        return self.tokenizer.apply_chat_template(conversation=messages, tokenize=False)
+
+    def create_error_response(
+        self,
+        message: str,
+        err_type: str = "BadRequestError",
+        status_code: HTTPStatus = HTTPStatus.BAD_REQUEST,
+    ) -> ErrorResponse:
+
+        error = Error(
+            message=message, type=err_type, param="", code=str(status_code.value)
+        )
+        return ErrorResponse(error=error)
+
+    def create_streaming_error_response(
+        self,
+        message: str,
+        err_type: str = "BadRequestError",
+        status_code: HTTPStatus = HTTPStatus.BAD_REQUEST,
+    ) -> ErrorResponse:
+        return self.create_error_response(
+            message=message, err_type=err_type, status_code=status_code
```

Comment:
What's the difference between error response and streaming error response?

Reply:
not required anymore. Moved to errors